src/frontend/src/optimizer/rule/over_agg_to_topn.rs

```diff
@@ -118,10 +122,15 @@ impl Rule for OverAggToTopNRule {
             }
         };
 
+        if offset > 0 && with_ties {
+            return None;
```

Comment:
Also unreachable here?

Reply:
It's unfortunately reachable when `RANK() > m`..